### ISSUE TITLE
Direct Downloads, secure?

### ISSUE THREAD

**Issue by zibelian** [akeeba/release-system#159]
Hi, 

Thanks for a great extension, i really love it. One thing i noticed, (i know, no support to be expected) but i dont know if this is a bug, security issue or just a functionality, as i cant find anything in the documentation. 

What i noticed is the following: Lest say i have the following structure: 

**Category**
demo catgeory

**Release**
demo release

**Items**
example1
example2
example3
etc... 

Category and releases acces are public but items access are set to registered users group, and to show unauthorised links.

Let's say i have enabled the direct link visibility and given the url of the item example1 to some registered users. Looking at the direct link url for the item example1 could look like this:
http://my-page/ars/demo-category/demo-category-demo-release/example-zip?format=zip&dlid=d2ade5beeb31725ff3b39303b5cf0064&jcompat=my.zip

Now, if i know the structure(category name, release name etc), which is not difficult to figure out, i would have theoretically unlimited download access to all items just by changing the item name in the url. Changing the name example1.zip to example2.zip in the url will allow me to download item2:
http://my-page/ars/demo-category/demo-category-demo-release/example2-zip?format=zip&dlid=d2ade5beeb31725ff3b39303b5cf0064&jcompat=my.zip

I also noticed that you can set whatever text u like after the item name in url and it will let you download the file, like for example:
http://my-page/ars/demo-category/demo-category-demo-release/example3_zip-whatever-i-like-here

There is a setting to remove the direct link visibility, but functionality is still active. I also understand that downloading of items can be set to only be allowed by certain group, but still, being part of that group means unlimited access to any item in that group using my examples above?

Have i missed something, or is it supposed to be like this. Could you please enlight me? Is there any way direct downloads can be completely deactivated?

Thank you

**Comment by nikosdion** [comment]
Obviously you have not read the fine manual.

The URL structure you "discovered" is actually [documented](https://github.com/akeeba/release-system/wiki/Understanding-the-structure-of-a-repository). So no surprises there.

Regarding the download ID, what you are describing is how the feature works. Download IDs are issued per user account and are, in fact, a substitute for a username and a password. That is, they are used to authenticate a user to the site. Therefore you should keep it secret. It should be fairly obvious, given the fact that a Download ID can be used to IDentify you to the Download system. See what I did there? That's why it's called a Download ID.

So, yeah, if you have PUBLISHED items which can be accessed by the user and the user knows how to get the URL to them they can append their Download ID to download them. This is not a security issue since the Download ID (only known to and visible by the user it belongs to) is an authentication token.

Remember: every logged in user sees a different Direct Download URL. The part after `dlid=` is their Main Download ID which is created automatically for them the first time it's needed. If you do not have a logged in user you'll see that there is no Direct Download URL. Guess why? Because without a logged in user there is no context to get a Download ID, therefore we cannot generate a direct URL.

Also remember: If the user did NOT have permissions to download the item or the item, release or category was unpublished then Download ID or not he would still not be able to download anything.

In conclusion the only problem reported here is that you did not read the documentation, you did not do your due diligence and you did not think this through.
